[WeeMingQing] iP

WeeMingQing opened the pull request:
# JonBird

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

## I am a bird bot to keep you on track of your tasks.

Here are my capabilities:
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use


All you need to do is:

1. download it from [here](https://github.com/WeeMingQing/ip/releases/tag/A-Jar).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you :laughing:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
***

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

WeeMingQing committed:
Rename, Greet, Exit.

WeeMingQing committed:
Add echo.

WeeMingQing committed:
Add list.

WeeMingQing committed:
Add Task class.

WeeMingQing committed:
Add inheritance classes.

WeeMingQing committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

* commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552':
  Bump gradle and lib version
  Add Gradle support

WeeMingQing committed:
Add TextUiTesting.

WeeMingQing committed:
Add exceptions.

WeeMingQing committed:
Add delete.

WeeMingQing committed:
Add enum.

WeeMingQing committed:
Add exception.

WeeMingQing committed:
Add out-of-range exception.

WeeMingQing committed:
Add Storage

WeeMingQing committed:
Add Storage

WeeMingQing committed:
Revert "Add Storage"

This reverts commit 3ff3b87e11a84d7fbccd10a70bc9a83e3a702a94.

WeeMingQing committed:
Merge branch 'master' into branch-Level-7

* master:
  Revert "Add Storage"

WeeMingQing committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add Storage

WeeMingQing committed:
Add dateTime object

WeeMingQing committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add dateTime object

WeeMingQing committed:
Add more OOP

WeeMingQing committed:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Add more OOP

WeeMingQing committed:
Refactor code

WeeMingQing committed:
Merge branch 'A-Packages'

* A-Packages:
  Refactor code

WeeMingQing committed:
Add gradle

WeeMingQing committed:
Merge branch 'A-Gradle'

* A-Gradle:
  Add gradle

WeeMingQing committed:
Add JUnit test

WeeMingQing committed:
Merge branch 'A-JUnit'

* A-JUnit:
  Add JUnit test

WeeMingQing committed:
Edit commands

WeeMingQing committed:
Add javadoc

WeeMingQing committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add javadoc

WeeMingQing committed:
Edit style

WeeMingQing committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit style

# Conflicts:
#	src/main/java/duke/task/Task.java

WeeMingQing committed:
Add find command

WeeMingQing committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find command

# Conflicts:
#	src/main/java/duke/task/Task.java

WeeMingQing committed:
Edit javadoc

yarnmengnus commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,94 @@
+
+    /**
+     * The constructor for AddCommand
+     * @param commandType The type of task being added; todo, deadline or event
```
Add an empty line between description and parameter

yarnmengnus replied:
Remember to add punctuation at the end of each param description

yarnmengnus commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,94 @@
+    public void execute(TaskList taskList, Storage storage) {
+        Task temp = null;
+        switch(commandType) {
+            case "T":
```
case shouldn't have indents

yarnmengnus commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,28 @@
+        System.out.println("JonBird:");
+        System.out.println("\tHere are the tasks in your list:");
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.println("\t\t"+ (i+1) + ". " + taskList.getTask(i).printTask());
```
whitespace

WeeMingQing committed:
Add checkstyle and edit style.

WeeMingQing committed:
Add GUI

WeeMingQing committed:
Merge branch 'GUI'

* GUI:
  Add GUI

Mahidharah commented on src/main/java/duke/task/Todos.java:
```diff
@@ -0,0 +1,23 @@
+package duke.task;
+
+/** Class to represent the task set by user*/
+public class Todos extends Task {
```
Class names should be singular if I am not wrong. Maybe change it to ToDo class instead. Nitpicking here

Mahidharah commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,196 @@
+     * @return Command
+     * @throws DukeException if the format of Command is wrong
+     */
+    public Command parse(String[] inp, TaskList taskList) throws DukeException {
```
Very long method, hard to follow. Consider using OOP to break them down into simpler steps?

Mahidharah reviewed (commented):
Just Nitpicking here. Overall neat code with correct style conventions!

tllshan commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,95 @@
+/**
+ * A class for the command for adding new task to list based on user input.
+ */
+public class AddCommand extends Command {
```
Class names should be nouns, if I'm not wrong. This would apply to the ByeCommand, DeleteCommand etc classes as well. Not too sure if this is applicable in your case though.

tllshan commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+public class FindCommand extends Command {
+
+    /** Keywords to search for in the task description*/
+    private final String keywords;
```
If I'm not wrong, I think plural form for variable names should only be used on a collection of objects? Perhaps this could be renamed as keyPhrase?

tllshan commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Override
+    public String printCommand(TaskList filteredTaskList) {
+        String result = "JonBird:\n\tHere are the tasks in your list:";
```
Perhaps using a StringBuilder for the result could be a better way to do this?

tllshan commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,111 @@
+        } catch (IOException e) {
+            //Throw DukeException
+        }
+
```
I think you missed this empty line, as well as on line 79 and 110, can remove them!

Nauxe commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,17 @@
+    public DukeException(String errorMsg) {
+        super(errorMsg);
+    }
+    @Override
```
```suggestion

@Override
```

tllshan commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,111 @@
+        String[] input = line.split(" \\| ");
+        if (input[0].equals("T")) {
+            Todos task = new Todos(input[2]);
+            if (input[1].equals("1")) {
```
I think you could consider defining a constant for "1", since you refer to it quite a few times in this method.
Eg "private static final String MARKED_AS_DONE = "1";", 
     then this line (and lines 97 & 103) can be rewritten as "if (input[1].equals(MARKED_AS_DONE)) {"

Nauxe commented on src/main/java/duke/main/DialogBox.java:
```diff
@@ -0,0 +1,62 @@
+    private Label dialog;
+    @FXML
+    private ImageView displayPicture;
+    /**
```
```suggestion

/**
```

tllshan reviewed (commented):
Code is generally well formatted and documented, good job! Just some slight nitpicking here and there to improve your code further.

Nauxe commented on src/main/java/duke/main/DateFormatter.java:
```diff
@@ -0,0 +1,74 @@
+     * @param dateTime The LocalDateTime containing dateTime from user input
+     * @return dateTime in String format
+     */
+    public String dateToString(LocalDateTime dateTime) {
```
Perhaps dateToString can be modified so you don't need the full string with hour and minute specifications, but can still input the hour and minutes if you want to

Nauxe reviewed (commented):
Looks good overall! Just some very slight formatting nitpicks and one potential suggestion.

WeeMingQing committed:
Refactor command parser

Current command parser method is too lengthy.

The method is split into smaller methods to perform specific parts of the function.

It is done to prevent long methods.

WeeMingQing committed:
Add assertions for Parser and Task class

The code correctness needs to be ensured at different point in time

Adding asserts in the different methods used for parsing user commands
allows us to check that our methods are handling them correctly

This is to ensure that the methods are doing their parts correctly

WeeMingQing committed:
Improve code quality

Current code violates certain code quality

The current code is not as readable and maintainable

Try to reduce long methods and ensure standard coding style

Using simpler methods and following standard coding style can improve readability and maintainability.

WeeMingQing committed:
Merge pull request #3 from WeeMingQing/branch-A-Assertions

Add assertions for Parser and Task class

WeeMingQing committed:
Merge branch 'master' of https://github.com/WeeMingQing/ip

* 'master' of https://github.com/WeeMingQing/ip:
  Add assertions for Parser and Task class

WeeMingQing committed:
Merge branch 'master' into branch-A-CodeQuality

WeeMingQing committed:
Merge pull request #4 from WeeMingQing/branch-A-CodeQuality

Improve code quality

WeeMingQing committed:
Merge branch 'master' of https://github.com/WeeMingQing/ip

* 'master' of https://github.com/WeeMingQing/ip:
  Improve code quality

WeeMingQing committed:
Add C-Undo extension

WeeMingQing committed:
Improve bot response and GUI appearance

WeeMingQing committed:
Merge pull request #5 from WeeMingQing/A-BetterGui

Improve bot response and GUI appearance

WeeMingQing committed:
Merge branch 'master' of https://github.com/WeeMingQing/ip

* 'master' of https://github.com/WeeMingQing/ip:
  Improve bot response and GUI appearance

WeeMingQing committed:
Fix box size for GUI

WeeMingQing committed:
Update GUI

WeeMingQing committed:
Add title

WeeMingQing committed:
Add User Guide

WeeMingQing committed:
Merge pull request #6 from WeeMingQing/add-user-guide

Add User Guide

WeeMingQing committed:
Merge branch 'master' of https://github.com/WeeMingQing/ip

* 'master' of https://github.com/WeeMingQing/ip:
  Add User Guide

WeeMingQing committed:
Final edits

WeeMingQing committed:
Final edits
